applicationset/controllers/applicationset_controller.go

```diff
@@ -164,7 +164,8 @@ func (r *ApplicationSetReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 				Status:  argov1alpha1.ApplicationSetConditionStatusTrue,
 			}, parametersGenerated,
 		)
-		return ctrl.Result{RequeueAfter: ReconcileRequeueOnValidationError}, err
+		// In order for the controller SDK to respect RequeueAfter, the error must be nil
+		return ctrl.Result{RequeueAfter: ReconcileRequeueOnValidationError}, nil
```

Comment:
Even though we set the error in the status do we also want to log an error? We do this here https://github.com/argoproj/argo-cd/blob/7cf2728a0a7ba43fac79e41b593ca2174fc36b9c/applicationset/controllers/applicationset_controller.go#L182

Reply:
Yea doesn’t hurt and won’t add any additional logs.